src/std/rpc/ws_client/client.rs

```diff
@@ -65,12 +68,12 @@ impl RpcClientTrait for WsRpcClient {
 			_ => json_req::author_submit_and_watch_extrinsic(&xthex_prefixed).to_string(),
 		};
 
-		let response = self.direct_rpc_request(jsonreq, SubmitAndWatchHandler::new(exit_on))?;
-		info!("Got response {} from waiting for {:?}", response, exit_on);
-		match exit_on {
-			XtStatus::Finalized | XtStatus::InBlock => Ok(Some(Hash::from_hex(response)?)),
-			XtStatus::Broadcast | XtStatus::Ready | XtStatus::SubmitOnly => Ok(None),
-			_ => Err(ApiClientError::UnsupportedXtStatus(exit_on)),
+		let maybe_response =
+			self.direct_rpc_request(jsonreq, SubmitAndWatchHandler::new(exit_on))??;
```

Comment:
Two `??` are necessary because two types of error may happen:
1.  ws connection error or channel error in the function `direct_rpc_request`
2. An Error of the `ThreadOut<ThreadMessage>`, e.g. an `RpcClientError`.